src/main/java/frc/trigon/robot/utilities/SpeedMechanism2d.java

```diff
@@ -0,0 +1,118 @@
+        this.currentVelocityBottomArrowLigament = currentVelocityLigament.append(new MechanismLigament2d(currentVelocityLigamentKeyZ + "CurrentVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kBlue)));
+
+        this.targetVelocityLigament = root.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityLigament", 0, 0, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityTopArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityTopArrowLigament", 0.2 * maxDisplayableVelocity, zeroTopAngle, 5, new Color8Bit(Color.kGray)));
+        this.targetVelocityBottomArrowLigament = targetVelocityLigament.append(new MechanismLigament2d(targetVelocityLigamentKeyZ + "TargetVelocityBottomArrowLigament", 0.2 * maxDisplayableVelocity, zeroBottomAngle, 5, new Color8Bit(Color.kGray)));
+    }
+
+    private boolean isCurrentVelocityLigamentBiggerThanTargetVelocityLigament(double currentVelocity, double targetVelocity) {
+        if (currentVelocity > deadband && targetVelocity > deadband)
+            return currentVelocity > targetVelocity;
+        else if (currentVelocity < -deadband && targetVelocity < -deadband)
+            return currentVelocity < targetVelocity;
+        return false;
+    }
+
+    private void setLigamentColor(double velocity) {
+        currentVelocityLigament.setColor(velocityToColor(velocity));
+        currentVelocityTopArrowLigament.setColor(velocityToColor(velocity));
+        currentVelocityBottomArrowLigament.setColor(velocityToColor(velocity));
+    }
+
+    private Color8Bit velocityToColor(double velocity) {
+        if (velocity > deadband)
+            return green;
+        else if (velocity < -deadband)
+            return red;
+        return blue;
+    }
+
+    private void setCurrentVelocityArrowAngle(double velocity) {
+        if (velocity > deadband) {
+            currentVelocityTopArrowLigament.setAngle(positiveTopAngle);
+            currentVelocityBottomArrowLigament.setAngle(positiveBottomAngle);
+        } else if (velocity < -deadband) {
+            currentVelocityTopArrowLigament.setAngle(negativeTopAngle);
+            currentVelocityBottomArrowLigament.setAngle(negativeBottomAngle);
+        }
+    }
+
+    private void setTargetVelocityArrowAngle(double targetVelocity) {
+        if (targetVelocity > deadband) {
+            targetVelocityTopArrowLigament.setAngle(positiveTopAngle);
+            targetVelocityBottomArrowLigament.setAngle(positiveBottomAngle);
+        } else if (targetVelocity < -deadband) {
+            targetVelocityTopArrowLigament.setAngle(negativeTopAngle);
+            targetVelocityBottomArrowLigament.setAngle(negativeBottomAngle);
+        }
+    }
```

Comment:
These are two very similar functions, extract them into one. Also, have one to update the position if the velocity is zero (below the deadband)